### PR TITLE
TEST: enable import tests for dataframes testing in `sklearnex.cluster.Kmeans`

### PR DESCRIPTION
**Pull request by samir-nasibli**
### Description 
Enabling CPU/GPU testing and with dataframes and queue for skleanrex.cluster.KMeans

- [x] I have reviewed my changes thoroughly before submitting this pull request.
- [x] I have commented my code, particularly in hard-to-understand areas.
- [ ] I have updated the documentation to reflect the changes, if necessary (updated in # - _add PR number_)
- [x] The unit tests pass successfully.
- [x] I have run it locally and tested the changes extensively.
- [x] I have resolved any merge conflicts that might occur with the base branch.
- [x] Git commit message contains an appropriate signed-off-by string _(see [CONTRIBUTING.md](https://github.com/intel/scikit-learn-intelex/blob/main/CONTRIBUTING.md#pull-requests) for details)_
- [x] I have added a respective label(s) to PR if I have a permission for that.

### CODE REVIEW

**Comment by icfaust** [comment]
KMeans yields an error from daal when attempting to GPU offload, and is getting enabled in #1770

**Comment by samir-nasibli** [comment]
> KMeans yields an error from daal when attempting to GPU offload, and this testing is getting enabled in #1770 along with GPU support

Thank you @icfaust !
I thought that we already have GPU offloading for Kmeans, just via daal4py backend. If it is buggy, I would like to highlight it somewhere. In tests deselection or some other way. Will think about it.

**Comment by md-shafiul-alam** [comment]
Also the test file has been updated in https://github.com/intel/scikit-learn-intelex/pull/1770. May be you can comment there if anything needs to be changed.

**Comment by samir-nasibli** [comment]
> Also the test file has been updated in #1770. May be you can comment there if anything needs to be changed.

I don't see any problem with enabling tests seperatly. Here just enabled dataframes testing. See [description ](https://github.com/intel/scikit-learn-intelex/pull/1888#issue-2369485107)

**Comment by samir-nasibli** [comment]
@md-shafiul-alam I would like to go with this changes.
This enablement includes not only GPU validation. Actually GPU validation could be addressed on #1832 or #1770, currently disabling this cases.
This PR makes kmeans enabled for dpnp/dpctl and other dataframes interop and Array API testing as well.
I hope this PR will be merged very soon, probably early that #1770

**Review by icfaust — COMMENTED** [review]
I have to agree with @md-shafiul-alam . This doesn't add any functional testing that isn't already there, and is actively under observation and development in #1770 . This also has some spelling mistakes/ informal language that would need to be addressed. I recommend closing this PR

**Comment by samir-nasibli** [comment]
> I have to agree with @md-shafiul-alam . This doesn't add any functional testing that isn't already there, and is actively under observation and development in #1770 . This also has some spelling mistakes/ informal language that would need to be addressed. I recommend closing this PR


Thanks for pointing out the typo errors. I will try to address them as soon as possible.

But I must express my disagreement with general opinion. I believe that the testing should cover the current implementation of the functionality. Currently we are declaring `numpy`/`dpctl` with CPU, `dpctl` with GPU, `dpnp` with CPU/`dpnp` with GPU/`pandas ` interop for our estimators. That should be covered.
I agree that, for now `Kmeans` case for the dpnp/dpctl with GPU sycl has some bug, that we are disabling.

Different type of fixes we have on PRs #1832 and #1770. I hope we will enable the test cases after #1832 or #1770 merge. I am also good with that #1770 will bring GPU enabling for such case and the test case itself will be updated.

**Comment by ethanglaser** [comment]
Agree with other comments, I don't see need for this. Shafi already added it with KMeans out of preview, and adding this validation in that PR makes more sense to me. Also comment in k_means.py will be outdated once KMeans out of preview is merged. I get that the timeline for it is not totally clear but I believe it will be merged soon enough that this isn't necessary.

**Comment by md-shafiul-alam** [comment]
I guess we can go ahead with this as the KMeans OOP can get pushed a little bit. Also might make sense to improve the coverage of tests if you feel this is in the scope of this PR.

**Comment by samir-nasibli** [comment]
@ethanglaser I need to express my disagreement with your opinion :)

Perhaps I didn’t convey the purpose of these changes well, but in this comment, I’ll try to provide a more substantial justification.

I am very pleased with the development on #1770 and I am confident that we will soon achieve the goals set there. However, I am sure that the goals of this PR and PR #1770 differ.
1) At the moment, there is functionality that is not validated by tests, and this PR aims to cover it. Kmeans has interop with dpnp/dpctl with CPU/GPU queue, numpy support, pandas interop (at least test is not failed). Planned to extend this test coverage for Array API support on #1861 and #1813.
2) Potential conflicts for #1770 can be resolved after merging this PR, and outdated code with comment will be completely removed. I do not see any problems here.

With this PR, I want to expand Kmeans testing on different dataframes. Changes of this kind have already been proposed at #1886, #1891 and integrated. This PR pursues the following the same logic.

Integration of this PR will also help to evaluate the support of Array API support, which is being done on PR

**Review by ethanglaser — APPROVED** [review]

**Comment by samir-nasibli** [comment]
Waiting Internal CI work done to continue merge.

**Comment by samir-nasibli** [comment]
Latest CI job: http://intel-ci.intel.com/ef38b90b-9b0a-f1aa-a67c-a4bf010d0e2e
